tm/src/main/java/com/alibaba/fescar/tm/api/transaction/TransactionHook.java

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.alibaba.fescar.tm.api.transaction;
+
+/**
+ * @author guoyao
+ * @date 2019/3/4
+ */
+public interface TransactionHook {
```

Comment:
As a transaction hook, why not also including before/after rollback?

Reply:
good idea .i will add it